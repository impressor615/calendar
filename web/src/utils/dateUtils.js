import moment from 'moment';

export const monthRanges = (date) => {
  const startDate = moment(date).startOf('month').startOf('week');
  const endDate = moment(date).endOf('month').endOf('week');
  const diff = endDate.diff(startDate, 'days') + 1;
  return Array.from({ length: diff }).map((_, index) => {
    const targetDate = moment(startDate).add(index, 'days');
    return {
      year: targetDate.year(),
      month: (targetDate.month() + 1),
      date: targetDate.date(),
      moment: targetDate,
    };
  });
};

export const weekRanges = (date) => {
  const startDate = moment(date).startOf('week');
  const endDate = moment(date).endOf('week');
  const diff = endDate.diff(startDate, 'days') + 1;
  return Array.from({ length: diff }).map((_, index) => {
    const targetDate = moment(startDate).add(index, 'days');
    return {
      year: targetDate.year(),
      month: (targetDate.month() + 1),
      date: targetDate.date(),
      moment: targetDate,
    };
  });
};
