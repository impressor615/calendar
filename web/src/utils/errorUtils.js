import CONSTANTS from '../constants';

const { ERRORS } = CONSTANTS;
export const getErrorMsg = (error) => {
  const { response } = error.request;
  const errorObj = JSON.parse(response);
  const msg = ERRORS[errorObj.code].ko || ERRORS['401'].ko;
  return msg;
}