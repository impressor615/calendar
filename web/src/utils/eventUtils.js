export const eventBind = (onEvent, props) => event => onEvent(event, props);