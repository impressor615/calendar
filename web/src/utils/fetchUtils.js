import axios from 'axios';


export const authAxios = axios.create({ headers: { 'x-access-token': process.env.REACT_APP_SECRET } });

export const fetchEvents = (startDate, endDate) => {
  const start_date = startDate.toISOString();
  const end_date = endDate.toISOString();
  return authAxios.get('/api/calendar', { params: { start_date, end_date }})
    .then(res => res.data)
};

export const updateServerEvent = (_id, putData, events) => (
  authAxios.put(`/api/calendar/${_id}`, putData)
    .then(() => {
      const newEvents = [...events].filter(item => item._id !== _id);
      putData._id = _id
      newEvents.push(putData);
      return newEvents;
    })
);

