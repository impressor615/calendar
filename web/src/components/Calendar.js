import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import { eventBind } from 'utils/eventUtils';
import CONSTANTS from '../constants';


export const EventBlock = ({ children, onToggle }) => (
  <button
    type="button"
    className="event-block btn"
    onClick={onToggle}
  >
    {children}
  </button>
);

const { DAY, HOURS } = CONSTANTS;
export const CalendarDay = ({
  day,
  dateObj,
  children,
  currentDate,
  onToggle,
}) => {
  const isYearShowing = currentDate.year() !== dateObj.year;
  const isMonthShowing = (currentDate.month() + 1) !== dateObj.month;
  return (
    <div
      className="cal-block"
      onClick={eventBind(onToggle, { type: 'month', dateObj })}
    >
      { day ? <div className="day">{day}</div> : null}
      <div>
        { isYearShowing ? `${dateObj.year}년 ${dateObj.month}월 ${dateObj.date}일` : null }
        { !isYearShowing && isMonthShowing ? `${dateObj.month}월 ${dateObj.date}일` : null }
        { !isYearShowing && !isMonthShowing ? `${dateObj.date}` : null }
      </div>
      {children}
    </div>
  );
}

export const MonthCalendar = ({ range, onToggle, currentDate, data }) => (
  <section className="month-calendar">
    {
      range.map((dateObj, index) => {
        const dataKey = dateObj.moment.format('YYYY-MM-DD');
        const events = data[dataKey];
        return (
          <CalendarDay
            key={index.toString()}
            currentDate={currentDate}
            dateObj={dateObj}
            day={DAY[index]}
            onToggle={onToggle}
          >
           {
              events
                ? (
                  <Fragment>
                    {
                      Object.entries(events).map(([key, event], index) => (
                        <EventBlock
                          key={index.toString()}
                          onToggle={eventBind(onToggle, { type: 'event', dateObj: event })}
                        >
                          &#9675; {`${key}:00 ${event.title}`}
                        </EventBlock>
                      ))
                    }
                  </Fragment>
                ) : null
            }
          </CalendarDay>
        );
      })
    }
  </section>
);

export const CalendarTime = ({ children, onToggle }) => (
  <button className="cal-block sm" onClick={onToggle}>
    {children}
  </button>
);

export const WeekCalendar = ({ range, onToggle, currentDate }) => (
  <section className="week-calendar">
    <div className="cal-block sm" />
    {
      range.map((dateObj, index) => {
        const isYearShowing = currentDate.year() !== dateObj.year;
        const isMonthShowing = (currentDate.month() + 1) !== dateObj.month;
        return (
          <div className="cal-block sm" key={index.toString()}>
            <div className="day">{DAY[index]}</div>
            <div>
              { isYearShowing ? `${dateObj.year}년 ${dateObj.month}월 ${dateObj.date}일` : null }
              { !isYearShowing && isMonthShowing ? `${dateObj.month}월 ${dateObj.date}일` : null }
              { !isYearShowing && !isMonthShowing ? `${dateObj.date}` : null }
            </div>
          </div>
        );
      })
    }
    {
      HOURS.map((hourObj, index) => (
        <Fragment key={index.toString()}>
          <div className="cal-block sm">
            <span className="hour">{hourObj.text}</span>
          </div>
          {
            range.map((dateObj, index) => (
              <CalendarTime
                key={index.toString()}
                onToggle={eventBind(onToggle, { type: 'week', hour: hourObj.code, dateObj })}
              />
            ))
          }
        </Fragment>
      ))
    }
  </section>
);

const Calendar = ({
  type,
  range,
  currentDate,
  onToggle,
  data,
}) => (
  <article className="calendar">
    {
      type === 'month'
        ? <MonthCalendar range={range} onToggle={onToggle} currentDate={currentDate} data={data} />
        : <WeekCalendar range={range} onToggle={onToggle} currentDate={currentDate} data={data} />
    }
  </article>
);

Calendar.propTypes = {
  type: PropTypes.string.isRequired,
  currentDate: PropTypes.object.isRequired,
  range: PropTypes.arrayOf(PropTypes.object).isRequired,
  data: PropTypes.object.isRequired,
  onToggle: PropTypes.func.isRequired,
};

export default Calendar;
