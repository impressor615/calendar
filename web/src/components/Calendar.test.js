import React from 'react';
import moment from 'moment';

import Calendar, {
  MonthCalendar,
  WeekCalendar,
  CalendarDay,
  CalendarTime,
  EventBlock,
} from 'components/Calendar';
import { weekRanges, monthRanges } from 'utils/dateUtils';


describe('<Calendar />', () => {
  const onToggle = jest.fn();
  const currentDate = moment('2018-10-13');
  const monthRange = monthRanges(currentDate);
  const weekRange = weekRanges(currentDate);
  let wrapper;
  const props = {
    type: 'month',
    currentDate,
    range: monthRange,
    onToggle,
    data: {
      '2018-10-13': {
        '9': {
          title: 'title',
          start_date: moment(currentDate).toISOString(),
          end_date: moment(currentDate).add(1, 'hours').toISOString(),
        },
      },
    },
  }

  afterEach(() => {
    wrapper.unmount();
  })

  it('should be rendered properly', () => {
    wrapper = mount(<Calendar {...props} />);
    expect(wrapper.find('.calendar')).toHaveLength(1);
    expect(wrapper.find(MonthCalendar)).toHaveLength(1);
    expect(wrapper.find(CalendarDay)).toHaveLength(props.range.length);
    expect(wrapper.find(EventBlock)).toHaveLength(1);

    wrapper.setProps({ type: 'week', range: weekRange });
    expect(wrapper.find(WeekCalendar)).toHaveLength(1);
    expect(wrapper.find(CalendarTime)).toHaveLength(24 * 7);
  });

  it('should be matched with the snapshot', () => {
    wrapper = mount(<Calendar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

