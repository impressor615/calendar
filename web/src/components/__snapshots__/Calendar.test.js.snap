// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> should be matched with the snapshot 1`] = `
<Calendar
  currentDate={"2018-10-12T15:00:00.000Z"}
  data={
    Object {
      "2018-10-13": Object {
        "9": Object {
          "end_date": "2018-10-12T16:00:00.000Z",
          "start_date": "2018-10-12T15:00:00.000Z",
          "title": "title",
        },
      },
    }
  }
  onToggle={[MockFunction]}
  range={
    Array [
      Object {
        "date": 30,
        "moment": "2018-09-29T15:00:00.000Z",
        "month": 9,
        "year": 2018,
      },
      Object {
        "date": 1,
        "moment": "2018-09-30T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 2,
        "moment": "2018-10-01T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 3,
        "moment": "2018-10-02T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 4,
        "moment": "2018-10-03T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 5,
        "moment": "2018-10-04T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 6,
        "moment": "2018-10-05T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 7,
        "moment": "2018-10-06T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 8,
        "moment": "2018-10-07T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 9,
        "moment": "2018-10-08T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 10,
        "moment": "2018-10-09T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 11,
        "moment": "2018-10-10T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 12,
        "moment": "2018-10-11T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 13,
        "moment": "2018-10-12T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 14,
        "moment": "2018-10-13T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 15,
        "moment": "2018-10-14T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 16,
        "moment": "2018-10-15T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 17,
        "moment": "2018-10-16T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 18,
        "moment": "2018-10-17T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 19,
        "moment": "2018-10-18T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 20,
        "moment": "2018-10-19T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 21,
        "moment": "2018-10-20T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 22,
        "moment": "2018-10-21T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 23,
        "moment": "2018-10-22T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 24,
        "moment": "2018-10-23T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 25,
        "moment": "2018-10-24T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 26,
        "moment": "2018-10-25T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 27,
        "moment": "2018-10-26T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 28,
        "moment": "2018-10-27T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 29,
        "moment": "2018-10-28T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 30,
        "moment": "2018-10-29T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 31,
        "moment": "2018-10-30T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      },
      Object {
        "date": 1,
        "moment": "2018-10-31T15:00:00.000Z",
        "month": 11,
        "year": 2018,
      },
      Object {
        "date": 2,
        "moment": "2018-11-01T15:00:00.000Z",
        "month": 11,
        "year": 2018,
      },
      Object {
        "date": 3,
        "moment": "2018-11-02T15:00:00.000Z",
        "month": 11,
        "year": 2018,
      },
    ]
  }
  type="month"
>
  <article
    className="calendar"
  >
    <MonthCalendar
      currentDate={"2018-10-12T15:00:00.000Z"}
      data={
        Object {
          "2018-10-13": Object {
            "9": Object {
              "end_date": "2018-10-12T16:00:00.000Z",
              "start_date": "2018-10-12T15:00:00.000Z",
              "title": "title",
            },
          },
        }
      }
      onToggle={[MockFunction]}
      range={
        Array [
          Object {
            "date": 30,
            "moment": "2018-09-29T15:00:00.000Z",
            "month": 9,
            "year": 2018,
          },
          Object {
            "date": 1,
            "moment": "2018-09-30T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 2,
            "moment": "2018-10-01T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 3,
            "moment": "2018-10-02T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 4,
            "moment": "2018-10-03T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 5,
            "moment": "2018-10-04T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 6,
            "moment": "2018-10-05T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 7,
            "moment": "2018-10-06T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 8,
            "moment": "2018-10-07T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 9,
            "moment": "2018-10-08T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 10,
            "moment": "2018-10-09T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 11,
            "moment": "2018-10-10T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 12,
            "moment": "2018-10-11T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 13,
            "moment": "2018-10-12T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 14,
            "moment": "2018-10-13T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 15,
            "moment": "2018-10-14T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 16,
            "moment": "2018-10-15T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 17,
            "moment": "2018-10-16T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 18,
            "moment": "2018-10-17T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 19,
            "moment": "2018-10-18T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 20,
            "moment": "2018-10-19T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 21,
            "moment": "2018-10-20T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 22,
            "moment": "2018-10-21T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 23,
            "moment": "2018-10-22T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 24,
            "moment": "2018-10-23T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 25,
            "moment": "2018-10-24T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 26,
            "moment": "2018-10-25T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 27,
            "moment": "2018-10-26T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 28,
            "moment": "2018-10-27T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 29,
            "moment": "2018-10-28T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 30,
            "moment": "2018-10-29T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 31,
            "moment": "2018-10-30T15:00:00.000Z",
            "month": 10,
            "year": 2018,
          },
          Object {
            "date": 1,
            "moment": "2018-10-31T15:00:00.000Z",
            "month": 11,
            "year": 2018,
          },
          Object {
            "date": 2,
            "moment": "2018-11-01T15:00:00.000Z",
            "month": 11,
            "year": 2018,
          },
          Object {
            "date": 3,
            "moment": "2018-11-02T15:00:00.000Z",
            "month": 11,
            "year": 2018,
          },
        ]
      }
    >
      <section
        className="month-calendar"
      >
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 30,
              "moment": "2018-09-29T15:00:00.000Z",
              "month": 9,
              "year": 2018,
            }
          }
          day="일"
          key="0"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div
              className="day"
            >
              일
            </div>
            <div>
              9월 30일
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 1,
              "moment": "2018-09-30T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          day="월"
          key="1"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div
              className="day"
            >
              월
            </div>
            <div>
              1
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 2,
              "moment": "2018-10-01T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          day="화"
          key="2"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div
              className="day"
            >
              화
            </div>
            <div>
              2
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 3,
              "moment": "2018-10-02T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          day="수"
          key="3"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div
              className="day"
            >
              수
            </div>
            <div>
              3
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 4,
              "moment": "2018-10-03T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          day="목"
          key="4"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div
              className="day"
            >
              목
            </div>
            <div>
              4
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 5,
              "moment": "2018-10-04T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          day="금"
          key="5"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div
              className="day"
            >
              금
            </div>
            <div>
              5
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 6,
              "moment": "2018-10-05T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          day="토"
          key="6"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div
              className="day"
            >
              토
            </div>
            <div>
              6
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 7,
              "moment": "2018-10-06T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="7"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              7
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 8,
              "moment": "2018-10-07T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="8"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              8
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 9,
              "moment": "2018-10-08T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="9"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              9
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 10,
              "moment": "2018-10-09T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="10"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              10
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 11,
              "moment": "2018-10-10T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="11"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              11
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 12,
              "moment": "2018-10-11T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="12"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              12
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 13,
              "moment": "2018-10-12T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="13"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              13
            </div>
            <EventBlock
              key="0"
              onToggle={[Function]}
            >
              <button
                className="event-block btn"
                onClick={[Function]}
                type="button"
              >
                ○ 
                9:00 title
              </button>
            </EventBlock>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 14,
              "moment": "2018-10-13T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="14"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              14
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 15,
              "moment": "2018-10-14T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="15"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              15
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 16,
              "moment": "2018-10-15T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="16"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              16
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 17,
              "moment": "2018-10-16T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="17"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              17
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 18,
              "moment": "2018-10-17T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="18"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              18
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 19,
              "moment": "2018-10-18T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="19"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              19
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 20,
              "moment": "2018-10-19T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="20"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              20
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 21,
              "moment": "2018-10-20T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="21"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              21
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 22,
              "moment": "2018-10-21T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="22"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              22
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 23,
              "moment": "2018-10-22T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="23"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              23
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 24,
              "moment": "2018-10-23T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="24"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              24
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 25,
              "moment": "2018-10-24T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="25"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              25
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 26,
              "moment": "2018-10-25T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="26"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              26
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 27,
              "moment": "2018-10-26T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="27"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              27
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 28,
              "moment": "2018-10-27T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="28"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              28
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 29,
              "moment": "2018-10-28T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="29"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              29
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 30,
              "moment": "2018-10-29T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="30"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              30
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 31,
              "moment": "2018-10-30T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            }
          }
          key="31"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              31
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 1,
              "moment": "2018-10-31T15:00:00.000Z",
              "month": 11,
              "year": 2018,
            }
          }
          key="32"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              11월 1일
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 2,
              "moment": "2018-11-01T15:00:00.000Z",
              "month": 11,
              "year": 2018,
            }
          }
          key="33"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              11월 2일
            </div>
          </div>
        </CalendarDay>
        <CalendarDay
          currentDate={"2018-10-12T15:00:00.000Z"}
          dateObj={
            Object {
              "date": 3,
              "moment": "2018-11-02T15:00:00.000Z",
              "month": 11,
              "year": 2018,
            }
          }
          key="34"
          onToggle={[MockFunction]}
        >
          <div
            className="cal-block"
            onClick={[Function]}
          >
            <div>
              11월 3일
            </div>
          </div>
        </CalendarDay>
      </section>
    </MonthCalendar>
  </article>
</Calendar>
`;
